### PR TITLE
feat: support GitLab Advisory Database

### PR DESCRIPTION
**Pull request by masahiro331**
https://github.com/aquasecurity/trivy/issues/940

### CODE REVIEW

File: pkg/vulnsrc/glad/glad_test.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+		},
+		{
+			name: "happy path maven",
+			dir:  filepath.Join("testdata", "happy"),
+			wantValues: []want{
+				{
+					key:   []string{"advisory-detail", "CVE-2018-1196", "maven::GitLab Advisory Database", "org.springframework.boot:spring-boot"},
+					value: `{"PatchedVersions":["1.5.10.RELEASE"],"VulnerableVersions":["(,1.5.10)"]}`},
+			},
+		},
```
Comment (knqyf263) [inline pkg/vulnsrc/glad/glad_test.go:43]: Looks like they can be aggregated.

File: pkg/vulnsrc/glad/glad_test.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		wantValues []want
+		wantErr    string
+	}{
+		{
+			name: "happy path go",
+			dir:  filepath.Join("testdata", "happy"),
+			wantValues: []want{
+				{
+					key:   []string{"advisory-detail", "CVE-2016-1905", "go::GitLab Advisory Database", "k8s.io/kubernetes"},
+					value: `{"PatchedVersions":["v1.2.0"],"VulnerableVersions":["\u003cv1.2.0"]}`,
```
Comment (knqyf263) [inline pkg/vulnsrc/glad/glad_test.go:31]: It might be better to define this value as a golden file.

File: pkg/vulnsrc/glad/glad_test.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+				},
+			},
+		},
+		{
+			name: "happy path maven",
+			dir:  filepath.Join("testdata", "happy"),
+			wantValues: []want{
+				{
+					key:   []string{"advisory-detail", "CVE-2018-1196", "maven::GitLab Advisory Database", "org.springframework.boot:spring-boot"},
+					value: `{"PatchedVersions":["1.5.10.RELEASE"],"VulnerableVersions":["(,1.5.10)"]}`},
```
Comment (knqyf263) [inline pkg/vulnsrc/glad/glad_test.go:41]: ditto

File: pkg/vulnsrc/glad/glad_test.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	tests := []struct {
+		name       string
+		dir        string
+		wantValues []want
+		wantErr    string
+	}{
+		{
+			name: "happy path go",
+			dir:  filepath.Join("testdata", "happy"),
+			wantValues: []want{
```
Comment (knqyf263) [inline pkg/vulnsrc/glad/glad_test.go:28]: We should test the `vulnerability-test` bucket as well.

Comment (masahiro331) [reply pkg/vulnsrc/glad/glad_test.go:29]: Is it different from vulnerability-detail?

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**